client/src/components/LightsOnOff.js

```diff
@@ -1,13 +1,73 @@
-import React, { useState, useRef } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import Button from '@material-ui/core/Button';
+import Box from '@material-ui/core/Box';
 import useUpdateServer from 'hooks/useUpdateServer';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import ErrorBlock from 'components/ErrorBlock'
 
-export default function LightsOnOff(props) {
+export default function LightsOnOff() {
   // constants
+  const STATE_URL = useRef(window.location.origin + '/state?component=power');
   const LIGHTS_UPDATE_URL = useRef(window.location.origin + '/lights');
+  const LIGHTSONOFF_FALLBACK_DEFAULT = useRef({
+    stringsOn: false,
+    grillOn: false
+  });
 
   // set up state
-  const [lightsPower, setLightsPower] = useState(props.initialState);
+  const [lightsPower, setLightsPower] = useState(LIGHTSONOFF_FALLBACK_DEFAULT.current);
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorMsgs, setErrorMsgs] = useState([]);
+
+  /**
+   * Adds an array of error strings to the existing state.
+   * @param {Array} errArray Each element is a user-facing error message string.
+   */
+  function updateErrors(errArray) {
+    if (!errArray.some( el => errorMsgs.includes(el))){ // don't duplicate errors. h/t https://stackoverflow.com/questions/16312528/check-if-an-array-contains-any-element-of-another-array-in-javascript
+      setErrorMsgs([...errorMsgs].concat(errArray))
+    }
+  }
+
+  // get state from server
+  useEffect(() => {    
```

Comment:
Echo async/await to make this less dense!

Reply:
as noted above will do in a future PR!